source/Halibut.Tests/TrustFixture.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut.Tests.Support;
+using Halibut.Tests.Support.TestAttributes;
+using Halibut.Tests.Support.TestCases;
+using Halibut.Tests.TestServices.Async;
+using Halibut.TestUtils.Contracts;
+using NUnit.Framework;
+
+namespace Halibut.Tests
+{
+    public class TrustFixture : BaseTest
+    {
+        [Test]
+        [LatestClientAndLatestServiceTestCases(testListening: true, testPolling: false, testWebSocket: false)]
+        public async Task TrustOnlyClosesConnectionsToUntrustedClients(ClientAndServiceTestCase clientAndServiceTestCase)
```

Comment:
This test doesn't test that connections are closed after a call to trustonly it tests that new requests can't be made to un-trusted services. That is a valuable test.

Is it worth adding a test which checks the connection is clossed? Perhaps with something like `WithCountTcpConnectionsCreated` or something similar (maybe counting closed connections)

Reply:
You can override the stream in the stream factory (in async) so you could also see the connection being closed via that. if that works better and you did want that test.

Reply:
I think this test covers the relevant functionality - we don't care specifically if the connection is closed, we just care that what was once trusted is no longer trusted (the connection itself is an implementation detail)